app/controllers/application_controller.rb

```diff
@@ -19,7 +19,12 @@ def set_locale
 
   private
     def extract_locale_from_accept_language_header
-      request.env['HTTP_ACCEPT_LANGUAGE'].scan(/^[a-z]{2}/).first
+      accept_language = request.env['HTTP_ACCEPT_LANGUAGE']
+      if accept_language.nil?
+        "en"
+      else
+        request.env['HTTP_ACCEPT_LANGUAGE'].scan(/^[a-z]{2}/).first
+      end
```

Comment:
This sets the locale to be the top-preferred locale of the user's browser, whether we have that locale translated or not.  Unfortunately, if the top-preferred locale is one we don't have, users see the translation key names in the page, rather than translated strings from an actual locale.

I think we should 
- Go through each of the user's preferred locales and use the first one we have.
- Find a way to revert to `en` when the user's requested locales aren't ones we have.

Can use a gem to handle this, or do it directly in custom Ruby logic.